lib/embulk/input/google_analytics/client.rb

```diff
@@ -122,7 +122,34 @@ def get_reports(page_token = nil)
         end
 
         def get_columns_list
-          get_custom_dimensions + get_metadata_columns
+          columns = get_custom_dimensions + get_metadata_columns
+          canonical_column_names(columns)
+        end
+
+        def canonical_column_names(columns)
+          result = []
+          columns.each do |col|
+            if col[:id].match(/XX$/)
+              # for such columns:
+              # https://developers.google.com/analytics/devguides/reporting/core/dimsmets#view=detail&group=content_grouping&jump=ga_contentgroupxx
+              min = [
+                col[:attributes][:minTemplateIndex],
+                col[:attributes][:premiumMinTemplateIndex],
+              ].compact.min
+              max = [
+                col[:attributes][:maxTemplateIndex],
+                col[:attributes][:premiumMaxTemplateIndex],
+              ].compact.max
+
+              min.upto(max) do |n|
+                actual_id = col[:id].gsub(/XX$/, n.to_s)
+                result << col.merge(id: actual_id)
```

Comment:
`col.merge(id: ...)` did it.
`col.merge` returns copy of `col` with overridden `:id`. So `attributes` or something keys are kept.